sql: TestDropColumnAfterMutations failed

sql.TestDropColumnAfterMutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3455503&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3455503&tab=artifacts#/) on master @ [23c5fba767da42cfcc01641ff0350a4dc0b6907a](https://github.com/cockroachdb/cockroach/commits/23c5fba767da42cfcc01641ff0350a4dc0b6907a):


```
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | 	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | runtime.goexit
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +Wraps: (2) failed to ingest index entries during backfill
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +Wraps: (3) Key (o)=(42) already exists.
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +Wraps: (4) constraint name: t_o_key
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +Wraps: (5) candidate pg code: 23505
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +Wraps: (6)
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | (opaque error wrapper)
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | reportable 0:
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  |
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | github.com/cockroachdb/cockroach/pkg/sql/row.NewUniquenessConstraintViolationError
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/row/errors.go:126
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*indexBackfiller).wrapDupError
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/indexbackfiller.go:403
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*indexBackfiller).ingestIndexEntries
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/indexbackfiller.go:302
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*indexBackfiller).runBackfill.func2
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/indexbackfiller.go:335
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | golang.org/x/sync/errgroup.(*Group).Go.func1
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | 	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | runtime.goexit
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +Wraps: (7) duplicate key value violates unique constraint "t_o_key"
I210916 18:55:26.034396 633081 jobs/registry.go:1150  [n1] 353 +Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *hintdetail.withDetail (4) *pgerror.withConstraintName (5) *pgerror.withCandidateCode (6) *errbase.opaqueWrapper (7) *errutil.leafError
W210916 18:55:26.042388 633081 sql/schema_changer.go:835  [n1,job=693856256892207105] 354  reversing schema change 693856256892207105 due to irrecoverable error: failed to ingest index entries during backfill: duplicate key value violates unique constraint "t_o_key"
I210916 18:55:26.044513 633081 sql/schema_changer.go:900  [n1,job=693856256892207105] 355  stepping through state machine
I210916 18:55:26.048538 633081 sql/schema_changer.go:985  [n1,job=693856256892207105] 356  finished stepping through state machine
    schema_changer_test.go:7345: query 'SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id])': expected:
        2
        
        got:
        3
        
W210916 18:55:26.063919 633186 kv/txn.go:687  [intExec=select-running/get-claimed-jobs] 357  failure aborting transaction: node unavailable; try another peer; abort caused by: connExecutor closing
W210916 18:55:26.064049 628173 jobs/registry.go:822  [-] 358  canceling all adopted jobs due to stopper quiescing
W210916 18:55:26.064275 632951 kv/kvserver/intentresolver/intent_resolver.go:790  [-] 359  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
E210916 18:55:26.064412 628175 jobs/registry.go:806  [-] 360  error processing claimed jobs: could not query for claimed jobs: context canceled
W210916 18:55:26.065564 627996 sql/sqlliveness/slinstance/slinstance.go:217  [n1] 361  exiting heartbeat loop
W210916 18:55:26.065816 632136 kv/txn.go:687  [n1,job=693856256657293313,scExec,id=54,mutation=1] 362  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
W210916 18:55:26.065402 629117 kv/txn.go:687  [n1,client=127.0.0.1:47842,hostssl,user=root] 363  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
W210916 18:55:26.066698 632599 kv/txn.go:687  [n1,client=127.0.0.1:47980,hostssl,user=root] 365  failure aborting transaction: node unavailable; try another peer; abort caused by: connExecutor closing
W210916 18:55:26.066739 628508 kv/txn.go:687  [n1,client=127.0.0.1:47834,hostssl,user=root] 366  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
W210916 18:55:26.067296 632136 sql/schema_changer.go:2141  [n1,job=693856256657293313] 367  error while running schema change, retrying: A dependent transaction failed for this schema change: job 693856256892207105 could not be loaded: context canceled
W210916 18:55:26.066431 631280 kv/txn.go:687  [n1,client=127.0.0.1:47898,hostssl,user=root] 364  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
W210916 18:55:26.069778 633081 kv/txn.go:687  [n1,job=693856256892207105] 368  failure aborting transaction: node unavailable; try another peer; abort caused by: failed to update running status of job 693856256892207105: job-update: context canceled
    --- FAIL: TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail (1.15s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestDropColumnAfterMutations PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-schema fqazi
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropColumnAfterMutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDropColumnAfterMutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3467581&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3467581&tab=artifacts#/) on master @ [d9278939d4be4d1109e9ed0a84c6e9cd2a8705bb](https://github.com/cockroachdb/cockroach/commits/d9278939d4be4d1109e9ed0a84c6e9cd2a8705bb):


```
=== RUN   TestDropColumnAfterMutations
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDropColumnAfterMutations479130109
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDropColumnAfterMutations
    schema_changer_test.go:7356: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDropColumnAfterMutations479130109
--- FAIL: TestDropColumnAfterMutations (1.84s)
=== RUN   TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail
    schema_changer_test.go:7345: query 'SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id])': expected:
        2
        
        got:
        3
        
    --- FAIL: TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail (0.53s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestDropColumnAfterMutations PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-schema fqazi
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropColumnAfterMutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDropColumnAfterMutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3484789&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3484789&tab=artifacts#/) on master @ [60b6d4d6e7c0cc30e991a8374ab9da1a0456c494](https://github.com/cockroachdb/cockroach/commits/60b6d4d6e7c0cc30e991a8374ab9da1a0456c494):


```
=== RUN   TestDropColumnAfterMutations
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDropColumnAfterMutations348601717
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDropColumnAfterMutations
    schema_changer_test.go:7356: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDropColumnAfterMutations348601717
--- FAIL: TestDropColumnAfterMutations (4.47s)
=== RUN   TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail
    schema_changer_test.go:7345: query 'SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id])': expected:
        2
        
        got:
        3
        
    --- FAIL: TestDropColumnAfterMutations/concurrent-drop-mutations-insert-fail (1.21s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestDropColumnAfterMutations PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-schema fqazi
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropColumnAfterMutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -7201,6 +7201,7 @@ COMMIT;
 			// This transaction will not complete. Therefore, we don't check the returned error.
 			_, _ = conn2.DB.ExecContext(context.Background(),
 				`
+	   SET application_name='TestDropColumnAfterMutations';
 	   BEGIN;
 	   ALTER TABLE t ALTER COLUMN j SET NOT NULL;
 	   ALTER TABLE t ADD COLUMN k INT8 DEFAULT 42 NOT NULL UNIQUE;
@@ -7215,6 +7216,7 @@ COMMIT;
 			// This transaction will not complete. Therefore, we don't check the returned error.
 			_, _ = conn1.DB.ExecContext(context.Background(),
 				`
+	   SET application_name='TestDropColumnAfterMutations';
 	   SET sql_safe_updates = false;
 	   BEGIN;
 	   ALTER TABLE t DROP COLUMN j;
@@ -7251,7 +7253,8 @@ COMMIT;
 		conn1.CheckQueryResults(t, fmt.Sprintf("SELECT status from system.jobs WHERE id = %d", jobIDs[0]), [][]string{{string(jobs.StatusRunning)}})
 		// Both jobs should be stuck in COMMIT, waiting for jobs to complete.
 		conn1.CheckQueryResults(t,
-			"SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id])",
+			"SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id]) "+
+				"AND application_name='TestDropColumnAfterMutations'",
 			[][]string{{"2"}},
 		)
 
@@ -7281,6 +7284,7 @@ COMMIT;
 			// This transaction will not complete. Therefore, we don't check the returned error.
 			_, _ = conn2.DB.ExecContext(context.Background(),
 				`
+	   SET application_name='TestDropColumnAfterMutations';
 	   BEGIN;
 	   ALTER TABLE t ALTER COLUMN j SET NOT NULL;
 	   ALTER TABLE t ADD COLUMN k INT8 DEFAULT 42 NOT NULL;
@@ -7293,6 +7297,7 @@ COMMIT;
 			// This transaction will not complete. Therefore, we don't check the returned error.
 			_, _ = conn1.DB.ExecContext(context.Background(),
 				`
+	   SET application_name='TestDropColumnAfterMutations';
 	   SET sql_safe_updates = false;
 	   BEGIN;
 	   ALTER TABLE t DROP COLUMN j;
@@ -7343,7 +7348,8 @@ COMMIT;
 		)
 		// Both jobs should be stuck in COMMIT, waiting for jobs to complete.
 		conn1.CheckQueryResults(t,
-			"SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id])",
+			"SELECT count(*) FROM [SHOW SESSIONS] WHERE last_active_query LIKE '%COMMIT%' AND session_id != (SELECT * FROM [SHOW session_id]) "+
+				"AND application_name='TestDropColumnAfterMutations'",
 			[][]string{{"2"}},
 		)
 

